src/__tests__/x509/sct.test.ts

```diff
@@ -0,0 +1,137 @@
+  );
+
+  describe('#parse', () => {
+    describe('when the SCT is valid', () => {
+      it('parses correctly', () => {
+        const sct = SignedCertificateTimestamp.parse(sctBuffer);
+
+        expect(sct.version).toEqual(0x00);
+        expect(sct.logID).toStrictEqual(Buffer.from(logID, 'hex'));
+        expect(sct.timestamp).toStrictEqual(Buffer.from(timestamp, 'hex'));
+        expect(sct.extensions).toHaveLength(0);
+        expect(sct.hashAlgorithm).toEqual(0x04);
+        expect(sct.signatureAlgorithm).toEqual(0x03);
+        expect(sct.signature).toStrictEqual(Buffer.from(signature, 'hex'));
+      });
+    });
+
+    describe('when the SCT is invalid', () => {
+      describe('when the SCT buffer is too short', () => {
+        it('throws an error', () => {
+          expect(() =>
+            SignedCertificateTimestamp.parse(Buffer.from(''))
+          ).toThrow('request past end of buffer');
+        });
+      });
+
+      describe('when the SCT buffer is too long', () => {
+        const sctBuffer = Buffer.from(
+          '00' +
+            logID +
+            timestamp +
+            '0000' +
+            '04' +
+            '03' +
+            '0047' +
+            signature +
+            'DEADBEEF', // extra bytes
+          'hex'
+        );
+
+        it('throws an error', () => {
+          expect(() => SignedCertificateTimestamp.parse(sctBuffer)).toThrow(
+            'SCT buffer length mismatch'
+          );
+        });
+      });
+    });
+  });
+
+  describe('#datetime', () => {
+    const subject = SignedCertificateTimestamp.parse(sctBuffer);
+
+    it('returns the parsed timestamp', () => {
+      expect(subject.datetime).toEqual(new Date('2022-07-22T21:11:51.196Z'));
+    });
+  });
+
+  describe('#algorithm', () => {
+    const subject = SignedCertificateTimestamp.parse(sctBuffer);
+
+    it('returns the hash algorithm', () => {
+      expect(subject.algorithm).toEqual('sha256');
+    });
+  });
+
+  describe('#verify', () => {
+    // Real pre-certificate used to generate the SCT
+    const preCert = Buffer.from(
+      'c355ee53d69e68aade04e5c6cc202cfac11fbcaa67e9a2ba7a64ced2aec8ccd000019b30820197a0030201020214466f689fbcc3be13e63ddbd14a277c41cd56d271300a06082a8648ce3d040303303731153013060355040a130c73696773746f72652e646576311e301c0603550403131573696773746f72652d696e7465726d656469617465301e170d3232303732323231313135315a170d3232303732323231323135315a30003059301306072a8648ce3d020106082a8648ce3d0301070342000459f2ab2bc2b2f9db98e711207b1c61d9f852fba456c40a1dcdd690de9ef9c2fba6133a565ccca76afdeea6352a1b0dbcf993e74e901892bca4ff3977a4a45410a381b73081b4300e0603551d0f0101ff04040302078030130603551d25040c300a06082b06010505070303301d0603551d0e041604141ce0136e2e5cdf1b097582a9326905ffc40f557f301f0603551d23041830168014dfd3e9cf56241196f9a8d8e92855a2c62e18643f301f0603551d110101ff041530138111627269616e40646568616d65722e636f6d302c060a2b0601040183bf300101041e68747470733a2f2f6769746875622e636f6d2f6c6f67696e2f6f61757468',
```

Comment:
What's up with the indentation here? Seems like the suggestion also get's weirdly formatted when I try to suggest an edit so maybe just UI bug or something?